lib/tabs/hourtable/pages/free_rooms.dart

```diff
@@ -155,3 +181,61 @@ Widget generateFreeRoomsClickDialog(BuildContext context, List<MapEntry<RoomType
     ],
   );
 }
+
+/// Dialog zum Auswählen der gewünschten anzuzeigenden Raumtypen
+class SetRoomTypeFilterDialog extends StatefulWidget {
+  const SetRoomTypeFilterDialog({super.key});
+
+  @override
+  State<SetRoomTypeFilterDialog> createState() => _SetRoomTypeFilterDialogState();
+}
+
+class _SetRoomTypeFilterDialogState extends State<SetRoomTypeFilterDialog> {
+  @override
+  Widget build(BuildContext context) {
+    final prefs = Provider.of<Preferences>(context);
+    return AlertDialog(
+      title: Text("Raumtypen ausblenden"),
+      content: SizedBox(
+        width: MediaQuery.sizeOf(context).width,
+        child: ListenableBuilder(
+          listenable: prefs,
+          builder: (ctx, _) => ListView(
+            padding: EdgeInsets.zero,
+            shrinkWrap: true,
+            children: RoomType.values.map((data) {
+              ListTile? genLT(RoomType roomTypeID) {
+                final String roomTypeIDString = roomTypeID.name;
+                return ListTile(
+                  contentPadding: EdgeInsets.only(left: 0.0, right: 0.0),
+                  leading: IconButton.outlined(
+                      icon: Icon(prefs.filteredRoomTypes.contains(roomTypeIDString) ? MdiIcons.eye : MdiIcons.eyeOff, size: 20),
+                      onPressed: () =>
+                        prefs.filteredRoomTypes.contains(roomTypeIDString)
+                            ? (prefs.removeFilteredRoomType(roomTypeID))
+                            : (prefs.addFilteredRoomType(roomTypeID))
+                  ),
+                  title: Text(
+                    (roomTypeID.toString() != "Allgemein") ? roomTypeID.toString() : "Allgemeine Räume",
+                   style: Theme.of(context).textTheme.bodyLarge,
+                  ),
+                );
+              }
+              final lt = genLT(data);
+              return lt;
```

Comment:
```suggestion
            children: RoomType.values.map((roomType) {
              return ListTile(
                contentPadding: EdgeInsets.only(left: 0.0, right: 0.0),
                leading: IconButton.outlined(
                    icon: Icon(prefs.filteredRoomTypes.contains(roomType) ? MdiIcons.eye : MdiIcons.eyeOff, size: 20),
                    onPressed: () =>
                      prefs.filteredRoomTypes.contains(roomType)
                          ? (prefs.removeFilteredRoomType(roomType))
                          : (prefs.addFilteredRoomType(roomType))
                ),
                title: Text(
                  (roomType == RoomType.unassigned) ? "Allgemeine Räume" : roomType.toString(),
                 style: Theme.of(context).textTheme.bodyLarge,
                ),
              );
```
Ich hab die Funktion `genLT` entfernt, weil die direkt hier aufgerufen wurde, die Variablen angepasst und den Enum-Vergleich mit Strings umgeändert.